src/sagemaker_training/mpi.py

```diff
@@ -334,6 +336,27 @@ def _create_command(self):
         for name in self._env_vars:
             command.extend(["-x", name])
 
+        def is_smddp_installed():
```

Comment:
The function name sound generic, whereas the function actually for torch specific smddp. Also, in the function you are looking for `smddmprun` binary. Please make the `is_smddp_installed` a property in the environment.py file ? 
and make use of `env.is_smddp_installed` .

Reply:
Updated it.